rest/rest_request_binding.h

```diff
@@ -397,12 +397,48 @@ createParameterExtractor(Json::Value & argHelp,
                 const RestRequest & request,
                 const RestRequestParsingContext & context)
         {
-            Json::Value parsed = request.payload == "" ?
-                Json::objectValue : Json::parse(request.payload);
+            Json::Value parsed = Json::parse(request.payload);
             return JsonCodec<T>::decode(p.name.empty() ? parsed : parsed[p.name]);
         };
 }
 
+template<typename T>
+static std::function<decltype(JsonCodec<T>::decode(std::declval<Json::Value>()))
+                     (RestConnection & connection,
+                      const RestRequest & request,
+                      const RestRequestParsingContext & context)>
+createParameterExtractor(Json::Value & argHelp,
+                         const JsonParamDefault<T> & p, void * = 0)
+{
+    Json::Value & v = argHelp["jsonParams"];
+    Json::Value & v2 = v[v.size()];
+    if (!p.name.empty()) {
+        v2["name"] = p.name;
+    }
+    v2["description"] = p.description;
+    v2["cppType"] = ML::type_name<T>();
+    v2["encoding"] = "JSON";
+    v2["location"] = "Request Body";
+
+    return [=] (RestConnection & connection,
+                const RestRequest & request,
+                const RestRequestParsingContext & context)
+        {
+            if (request.payload == "") {
+                return p.defaultValue;
+            }
+            Json::Value parsed = Json::parse(request.payload);
+            if (p.name.empty()) {
+                return JsonCodec<T>::decode(parsed);
+            }
+            if (parsed.isMember(p.name.rawString())) {
+                return JsonCodec<T>::decode(parsed[p.name]);
```

Comment:
Because the original jsoncpp doesn't know about Datacratic utf8 strings and we've only added overrides as needed.  It should be added to the interface.